packages/shared/lib/services/sync/run.service.ts

```diff
@@ -24,7 +24,28 @@ import type { Environment } from '../../models/Environment';
 import type { Metadata } from '../../models/Connection';
 import * as recordsService from './data/records.service.js';
 
+interface BigQueryClientInterface {
+    insert(row: RunScriptRow): void;
+}
+
+interface RunScriptRow {
+    executionType: string;
+    internalConnectionId: number | undefined;
+    connectionId: string;
+    accountId: number | undefined;
+    scriptName: string;
+    scriptType: string;
+    environmentId: number;
+    providerConfigKey: string;
+    status: string;
+    syncId: string;
+    content: string;
+    runTimeInSeconds: number;
+    createdAt: number;
+}
+
 interface SyncRunConfig {
+    bigQueryClient: BigQueryClientInterface;
```

Comment:
can this be optional so we don't even need to pass it in the tests or any situation it is no required?

Reply:
done